src/main/java/com/bravo/user/controller/PaymentController.java

```diff
@@ -0,0 +1,36 @@
+package com.bravo.user.controller;
+
+import com.bravo.user.annotation.SwaggerController;
+import com.bravo.user.model.dto.PaymentDto;
+import com.bravo.user.service.PaymentService;
+import com.bravo.user.service.UserService;
+import com.bravo.user.utility.PageUtil;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+
+@RequestMapping(value = "/payment")
+@SwaggerController
+public class PaymentController {
+    private final PaymentService paymentService;
+
+    public PaymentController(PaymentService paymentService) {
+        this.paymentService = paymentService;
+    }
+
+    @GetMapping(value = "/retrieve/{userId}")
+    @ResponseBody
+    public List<PaymentDto> retrieve(
+            final @PathVariable String userId,
+            final @RequestParam(required = false) Integer page,
+            final @RequestParam(required = false) Integer size,
+            final HttpServletResponse httpResponse
+    ) {
+            final PageRequest pageRequest = PageUtil.createPageRequest(page, size);
+            return paymentService.retrieveByUserId(userId, pageRequest, httpResponse);
```

Comment:
Is it safe to use the `userId` as is? 😨

Reply:
No, but the instructions were to use a userId as a @PathVariable, I would prefer to use a JWT and pull a unique identifier from that and retrieve user data from some kind of data store.